marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/DhsDeployer.java

```diff
@@ -0,0 +1,182 @@
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.deploy.HubAppDeployer;
+import com.marklogic.hub.deploy.commands.DeployDatabaseFieldCommand;
+import com.marklogic.hub.deploy.commands.HubDeployDatabaseCommandFactory;
+import com.marklogic.hub.deploy.commands.LoadUserArtifactsCommand;
+import com.marklogic.hub.deploy.commands.LoadUserModulesCommand;
+import com.marklogic.hub.impl.HubConfigImpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Deploys resources, modules, and schemas to a DHS instance. Assumes that the user used to deploy these objects has
+ * the "data-hub-developer" role.
+ */
+public class DhsDeployer extends LoggingObject {
+
+    public void deployToDhs(HubConfigImpl hubConfig) {
+        prepareAppConfigForDeployingToDhs(hubConfig);
+
+        HubAppDeployer dhsDeployer = new HubAppDeployer(hubConfig.getManageClient(), hubConfig.getAdminManager(), null, hubConfig.newStagingClient());
+        dhsDeployer.setCommands(buildCommandListForDeployingToDhs(hubConfig));
+        dhsDeployer.deploy(hubConfig.getAppConfig());
+    }
+
+    /**
+     * The AppConfig object belonging to HubConfig can be configured based on known DHS environment settings. This
+     * minimizes the number of properties that a user needs to specify themselves - i.e. the user shouldn't be expected
+     * to configure all the properties that are known to be true about a DHS instance.
+     *
+     * @param hubConfig
+     */
+    protected void prepareAppConfigForDeployingToDhs(HubConfig hubConfig) {
+        setKnownValuesForDhsDeployment(hubConfig);
+
+        AppConfig appConfig = hubConfig.getAppConfig();
+
+        // We always want all user modules loaded
+        appConfig.setModuleTimestampsPath(null);
+
+        // We never want forests created, DHS handles that
+        appConfig.setCreateForests(false);
+
+        removeHubInternalConfigDir(appConfig);
+
+        // 8000 is not available in DHS
+        int port = hubConfig.getPort(DatabaseKind.STAGING);
+        logger.info("Setting App-Services port to: " + port);
+        appConfig.setAppServicesPort(port);
+
+        if (hubConfig.getSimpleSsl(DatabaseKind.STAGING)) {
+            logger.info("Enabling simple SSL for App-Services");
+            appConfig.setAppServicesSimpleSslConfig();
+        }
+
+        String authMethod = hubConfig.getAuthMethod(DatabaseKind.STAGING);
+        if (authMethod != null) {
+            logger.info("Setting security context type for App-Services to: " + authMethod);
+            appConfig.setAppServicesSecurityContextType(SecurityContextType.valueOf(authMethod.toUpperCase()));
+        }
+    }
+
+    /**
+     * The contents of hub-internal-config should have been deployed to DHS already via the DHF installer. And
+     * it is not expected that a user will not have permission to deploy all of the resources in this directory.
+     *
```

Comment:
can we make the comment simpler. "not expected" followed by "not have" is confusing and makes me re-read it.

Reply:
Also the method name here implies that we are removing hub-internal-config. But we only are ignoring it for dhsDeploy.

Reply:
Good points, I made the updates.